relayer/app/cursor/stream.go

```diff
@@ -0,0 +1,137 @@
+package cursor
+
+import (
+	"context"
+	"math"
+	"slices"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/xchain"
+)
+
+type stream struct {
+	SrcVersion xchain.ChainVersion
+	DestChain  netconf.Chain
+	Network    netconf.Network
+}
+
+// getAllStreamCursors returns all cursors from the storage.
+//
+// Cursors are ordered by source chain ID, confirmation level and destination chain ID in ascending order.
+func getAllStreamCursors(ctx context.Context, db CursorTable, stream stream) ([]*Cursor, error) {
+	iterator, err := db.List(ctx,
+		CursorPrimaryKey{}.WithSrcChainIdConfLevelDstChainId(
+			stream.SrcVersion.ID,
+			uint32(stream.SrcVersion.ConfLevel),
+			stream.DestChain.ID,
+		))
+
+	if err != nil {
+		return nil, errors.Wrap(err, "list cursors")
+	}
+	defer iterator.Close()
+
+	var cursors []*Cursor
+	for iterator.Next() {
+		cursor, err := iterator.Value()
+		if err != nil {
+			return nil, errors.Wrap(err, "cursor value")
+		}
+		cursors = append(cursors, cursor)
+	}
+
+	return cursors, nil
+}
+
+// ConfirmStream checks all stream cursors and confirms them according to the rules:
+//   - if a cursor is not empty fetch finalized cursor from the network and confirm only if finalized
+//     message offset is higher to the cursor last message offset
+//   - if a cursor is empty, confirm it if the previous cursor was confirmed
+func ConfirmStream(ctx context.Context, db CursorTable, network netconf.Network, xProvider xchain.Provider, stream stream) error {
```

Comment:
nit: don't export types and fields by default, rather make them private by default